source/slang/slang-lower-to-ir.cpp

```diff
@@ -5170,7 +5170,11 @@ struct DeclLoweringVisitor : DeclVisitor<DeclLoweringVisitor, LoweredValInfo>
                         // Need to construct a sub-witness-table
                         auto irWitnessTableBaseType = lowerType(subContext, astReqWitnessTable->baseType);
                         irSatisfyingWitnessTable = subBuilder->createWitnessTable(irWitnessTableBaseType);
-
+                        auto mangledName = getMangledNameForConformanceWitness(
+                            subContext->astBuilder,
+                            astReqWitnessTable->witnessedType,
+                            astReqWitnessTable->baseType);
+                        subBuilder->addExportDecoration(irSatisfyingWitnessTable, mangledName.getUnownedSlice());
```

Comment:
The second issue I'm pointing out here is more closely related to a statement in the PR description: it is mentioned that this change makes it so that even the witness tables for associated types get mangled names (and this logic seems to be the part that is doing that...).

I don't think that is correct/necessary.

If the only way a witness table is referenced is as an entry in another witness table, then there is no need for the child witness table to have an `[export(...)]` decoration and linkage.

I suspect that the logic here in IR lowering should be backed out, and we should retain the current rules for when declarations get an `[export(...)]` decoration as-is.

Instead, the emit logic should pay attention to whether a global-scope entity (like a witness table) has linkage, and use that to decide between emitting it as `extern "C"` with the mangled name, or as a `static` with a synthesized name).

I don't like the idea of forcing things to have linkage that don't actually need it, just to simplify our code generation.